File: spring2013/codeWar/LRS/Clients/PlayerCSharpAI/api/FlagState.cs
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Xml.Linq;

namespace PlayerCSharpAI.api
{
	/// <summary>
	/// The state of the flags for a player.
	/// </summary>
	public class FlagState
	{
		/// <summary>
		/// Build from XML.
		/// </summary>
		/// <param name="element">Initialize with the values in this object.</param>
		public FlagState(XElement element)
		{
			Position = new Point(int.Parse(element.Attribute("x").Value), int.Parse(element.Attribute("y").Value));
			Touched = element.Attribute("touched").Value.ToLower() == "true";
		}

		public static List<FlagState> FromXML(XElement element)
		{
			return element.Elements("flag-state").Select(flag => new FlagState(flag)).ToList();
		}

		/// <summary>
		/// Which square the flag is on.
		/// </summary>
		public Point Position { get; protected set; }

		/// <summary>
		/// true if the flag has been touched. Once all 3 flags are touched a player has won.
		/// </summary>
		public bool Touched { get; set; }
	}
}

File: spring2013/codeWar/LRS/Clients/PlayerCppAI/FlagState.cpp
#include "FlagState.h"

namespace PlayerCSharpAI
{
	namespace api
	{

		FlagState::FlagState(const xml_node& element)
		{
			m_pos = Point(element.attribute("x").as_int(), element.attribute("y").as_int());
			m_touched = element.attribute("touched").as_bool();
		}

		vector<FlagState>* FlagState::FromXML(const xml_node& element)
		{
			vector<FlagState>* v = new vector<FlagState>;
			for (xml_node ndFlgSt = element.child("flag-state"); ndFlgSt; ndFlgSt = ndFlgSt.next_sibling("flag-state"))
				v->push_back(FlagState(ndFlgSt));
			return v;
		}

		Point FlagState::getPosition() const
		{
			return m_pos;
		}

		void FlagState::setPosition(Point value)
		{
			m_pos = value;
		}

		bool FlagState::getTouched() const
		{
			return m_touched;
		}

		void FlagState::setTouched(bool value)
		{
			m_touched = value;
		}
	}
}
